[Modularization] SofaDenseSolver

fredroy opened the pull request:
NOTE: the **template="NewMat"** for both CGLinearSolver CholeskyLinearSolver will now NEED the package SofaDenseSolver.

#1298 



______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

fredroy committed:
Modularize SofaDenseSolver (+cleaning)

fredroy committed:
removing remnants from other modules

fredroy committed:
Fix linking problem

fredroy committed:
Change _API keyword for more consistency

fredroy committed:
FIX Sofa_Target replacement

fredroy commented:
[ci-build][with-all-tests]

guparan commented on modules/SofaDenseSolver/SofaDenseSolverConfig.cmake.in:
```diff
@@ -0,0 +1,13 @@
+
+sofa_find_package(Newmat REQUIRED)
+sofa_find_package(SofaBase REQUIRED)
+
```
Do not use `sofa_find_package` in Config.cmake, it needs SofaMacros.cmake and it's not sure the caller already has it.

damienmarchal commented on modules/CMakeLists.txt:
```diff
@@ -5,8 +5,13 @@ sofa_add_module(SofaSparseSolver SofaSparseSolver ON)      #
+
 if(SOFA_NO_OPENGL)
-    message("SOFA_NO_OPENGL flag prevents from using the SofaCUDA, SofaSimpleGUI and VolumetricRendering plugins")
+    message("SOFA_NO_OPENGL flag prevents from using the SofaOpenglVisual plugin")
```
Not sure.

fredroy replied:
For SofaCUDA, sure ;) I did a PR to remove the need of OpenGL
By the way, for me we should not show the message of support/not support at this place. It should be where SofaSimpleGUI is added

fredroy committed:
Missing required standard init*() for plugin, and add ComponentChange message

fredroy committed:
Add requiredplugin in scenes with sofadensesolver component

fredroy committed:
Get plugin version from Cmake directly

fredroy commented:
[ci-build][force-full-build]

fredroy committed:
Merge branch 'master' of https://github.com/sofa-framework/sofa into modularize_sofadensesolver

fredroy commented:
[ci-build][with-all-tests]

fredroy commented on modules/SofaDenseSolver/src/SofaDenseSolver/initDenseSolver.cpp:
```diff
@@ -19,35 +19,39 @@
 {
 
 namespace linearsolver
 {
```
the namespace linearsolver is needed only for the explicit instanciation of MatrixLinearSolver with the NewMat template

fredroy commented on modules/SofaDenseSolver/README.md:
```diff
@@ -0,0 +1,7 @@
+Plugin with Solvers using dense matrices.
+Needs NewMat.
+
+
```
can do

fredroy committed:
Merge branch 'master' of https://github.com/sofa-framework/sofa into modularize_sofadensesolver

fredroy committed:
Update Readme

guparan commented on SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp:
```diff
@@ -109,6 +109,10 @@ std::map<std::string, ComponentChange> uncreatableComponents = {
+    // SofaDenseSolver was pluginized in #1299
+    {"LULinearSolver", Pluginized("v20.06", "SofaDenseSolver")},
+    //{"NewMatCholeskySolver", Pluginized("v20.06", "SofaDenseSolver")},
+    //{"NewMatCGLinearSolver", Pluginized("v20.06", "SofaDenseSolver")},
```
Why commented?

guparan replied:
Yes, that's the point of adding these two lines. Still don't understand why they are commented out though.

fredroy replied:
No aliases (I think you cannot aliase a SofaObject on a particular template by the way)
The problem of ComponentChange is that you cannot deprecate a SofaObject on a particular template, e.g it is no possible to deprecate only the NewMat template of CGLinearSolver.
So I let these two lines commented (as NewMatCholeskySolver and NewMatCGLinearSolver do not exist) to keep the deprecations in mind or if ComponentChange handles template in the future.
Of course, I can just remove these two lines anyway if it is bothering.

guparan committed:
[CMake] FIX SofaAdvanced package deps

Soft dependencies like "SofaNonUniformFem -> SofaDenseSolver" must be handled at CMake package level too.

guparan commented:
This PR has been merged but status was not updated due to GitHub service degradation.
